ui/src/turing/components/form/standard_ensembler/LinkedRoutesTable.js

```diff
@@ -0,0 +1,166 @@
+import React, { useEffect, useMemo, useState } from "react";
+
+import {
+  EuiFlexItem,
+  EuiLoadingChart,
+  EuiTextAlign,
+  EuiInMemoryTable,
+  EuiIcon,
+  EuiTextColor,
+} from "@elastic/eui";
+
+import { useXpApi } from "hooks/useXpApi";
+import moment from "moment";
+import { useConfig } from "config";
+import { getExperimentStatus } from "services/experiment/ExperimentStatus";
+import { LinkedExperimentsContextMenu } from "./LinkedExperimentsContextMenu";
+
+export const LinkedRoutesTable = ({
+  projectId,
+  routes,
+  treatmentConfigRouteNamePath,
+}) => {
+  const { appConfig } = useConfig();
+
+  const [isButtonPopoverOpen, setIsButtonPopoverOpen] = useState(routes.reduce((m, r) => {m[r.id] = {running: false, scheduled: false}; return m}, {}));
```

Comment:
Hmmm having this state in the parent component makes the implementation a little more complex and less coupled with the button that it belongs to. Can this be defined directly inside `LinkedExperimentsContextMenu` component so that each clickable item has its own state? And can it simply track `true/false` ? Then, if any of them are true, we can use the `routeToExperimentMappings` to list the experiments in the popover, for that specific status - right?

We will be creating more state variables but since they are all lean, I don't think that should be a problem.

Reply:
I remember making a conscious choice to manage all the states of the parent because creating the individual states within `LinkedExperimentsContextMenu` somehow threw some error or minimally a warning regarding the creation of individual states by looping through with the function in `render`. I'll try to reproduce the issue again and share that error again.

Reply:
Okay I can't seem to get the error to reappear, maybe it was something else :/ I've made the changes to track the state of each button within each `LinkedExperimentsContextMenu`, and we'll be using `useToggle` to get the `true/false` state of the button.